src/backend/llvm.py

```diff
@@ -195,38 +207,79 @@ def gen_function_prototype(self, fn):
         else:
             # Otherwise create a new function
             func = ir.Function(self.module, func_ty, funcname)
-        # Set function argument names from AST
-        for i, arg in enumerate(func.args):
-            arg.name = fn.args[i].name
-            self.func_symtab[fn.args[i].name] = arg
 
         return func
 
+    def declare_var (self, typ, name):
+        """Allocate a pointer using alloc and add it to the symbol table, if it doesn't already exist"""
+        #if name in self.func_alloca_symtab: return
+        if name in self.func_alloca_symtab or self.func_void_syms: return
+
+        if isinstance(typ, ir.VoidType):
+            self.func_void_syms [name] = None
+        else:
+            builder = ir.IRBuilder (self.func_bbs ['alloca']) # Use a separate builder so we don't mess with the global builder's position
```

Comment:
create a random string instead of alloca because program might be using this label